### PR TITLE
DROOLS-6571 local reproducer

### PR DESCRIPTION
**Pull request by tarilabs**
**JIRA**: https://issues.redhat.com/browse/DROOLS-6571

**referenced Pull Requests**: follow-up on:

* https://github.com/kiegroup/drools/pull/3803

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

* <b>a pull request</b> please add comment: <b>Jenkins retest this</b>
 
* <b>a full downstream build</b> please add comment: <b>Jenkins run fdb</b>
  
* <b>a compile downstream build</b> please  add comment: <b>Jenkins run cdb</b>

* <b>a full production downstream build</b> please add comment: <b>Jenkins execute product fdb</b>

* <b>an upstream build</b> please add comment: <b>Jenkins run upstream</b>
</details>

### CODE REVIEW

**Comment by tarilabs** [comment]
I am using this PR to demonstrate "locally" to Drools the issue without relying on the kie-server reproducer, thanks to help from @mariofusco 
(before merging I would naturally keep the fix in the related PR)

**Comment by tarilabs** [comment]
demonstrated "locally" to Drools repo with https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/main/job/pullrequest/job/drools-main.pr/186/

image

**Comment by tarilabs** [comment]
CDB failure is unrelated

```
[2021-09-01T14:44:20.806Z] 1) Error in custom provider, java.lang.TypeNotPresentException: Type io.quarkus.maven.components.BootstrapSessionListener not present
[2021-09-01T14:44:20.806Z]   at ClassRealm[extension>io.quarkus:quarkus-maven-plugin:2.2.0.CR1, parent: sun.misc.Launcher$AppClassLoader@7852e922] (via modules: org.eclipse.sisu.wire.WireModule -> org.eclipse.sisu.plexus.PlexusBindingModule)
[2021-09-01T14:44:20.806Z]   while locating org.apache.maven.AbstractMavenLifecycleParticipant annotated with @com.google.inject.name.Named(value=quarkus-bootstrap)
[2021-09-01T14:44:20.806Z] 
[2021-09-01T14:44:20.806Z] 1 error
[2021-09-01T14:44:20.806Z]     at com.google.inject.internal.InternalProvisionException.toProvisionException (InternalProvisionException.java:226)
[2021-09-01T14:44:20.806Z]     at com.google.inject.internal.InjectorImpl$1.get (InjectorImpl.java:1053)
[2021-09-01T14:44:20.806Z]     at org.eclipse.sisu.inject.LazyBeanEntry.getValue (LazyBeanEntry.java:81)
[2021-09-01T14:44:20.806Z]     at org.eclipse.sisu.plexus.LazyPlexusBean.getValue (LazyPlexusBean.java:51)
[2021-09-01T14:44:20.806Z]     at org.eclipse.sisu.wire.EntryListAdapter$ValueIterator.next (EntryListAdapter.java:111)
[2021-09-01T14:44:20.806Z]     at java.util.AbstractCollection.addAll (AbstractCollection.java:343)
[2021-09-01T14:44:20.806Z]     at org.apache.maven.DefaultMaven.getLifecycleParticipants (DefaultMaven.java:397)
[2021-09-01T14:44:20.806Z]     at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:260)
[2021-09-01T14:44:20.806Z]     at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
[2021-09-01T14:44:20.806Z]     at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
[2021-09-01T14:44:20.806Z]     at org.apache.maven.cli.MavenCli.execute (MavenCli.java:957)
[2021-09-01T14:44:20.806Z]     at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:289)
[2021-09-01T14:44:20.806Z]     at org.apache.maven.cli.MavenCli.main (MavenCli.java:193)
[2021-09-01T14:44:20.806Z]     at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
[2021-09-01T14:44:20.806Z]     at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
[2021-09-01T14:44:20.806Z]     at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
[2021-09-01T14:44:20.806Z]     at java.lang.reflect.Method.invoke (Method.java:498)
[2021-09-01T14:44:20.806Z]     at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
[2021-09-01T14:44:20.806Z]     at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
[2021-09-01T14:44:20.806Z]     at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
[2021-09-01T14:44:20.806Z]     at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
[2021-09-01T14:44:20.806Z] Caused by: java.lang.TypeNotPresentException: Type io.quarkus.maven.components.BootstrapSessionListener not present
[2021-09-01T14:44:20.806Z]     at org.eclipse.sisu.space.URLClassSpace.loadClass (URLClassSpace.java:147)
[2021-09-01T14:44:20.806Z]     at org.eclipse.sisu.space.NamedClass.load (NamedClass.java:46)
[2021-09-01T14:44:20.806Z]     at org.eclipse.sisu.space.AbstractDeferredClass.get (AbstractDeferredClass.java:48)
[2021-09-01T14:44:20.806Z]     at com.google.inject.internal.ProviderInternalFactory.provision (ProviderInternalFactory.java:85)
[2021-09-01T14:44:20.806Z]     at com.google.inject.internal.InternalFactoryToInitializableAdapter.provision (InternalFactoryToInitializableAdapter.java:57)
[2021-09-01T14:44:20.806Z]     at com.google.inject.internal.ProviderInternalFactory$1.call (ProviderInternalFactory.java:66)
[2021-09-01T14:44:20.806Z]     at com.google.inject.internal.ProvisionListenerStackCallback$Provision.provision (ProvisionListenerStackCallback.java:112)
[2021-09-01T14:44:20.807Z]     at org.eclipse.sisu.bean.BeanScheduler$CycleActivator.onProvision (BeanScheduler.java:230)
[2021-09-01T14:44:20.807Z]     at com.google.inject.internal.ProvisionListenerStackCallback$Provision.provision (ProvisionListenerStackCallback.java:120)
[2021-09-01T14:44:20.807Z]     at com.google.inject.internal.ProvisionListenerStackCallback.provision (ProvisionListenerStackCallback.java:66)
[2021-09-01T14:44:20.807Z]     at com.google.inject.internal.ProviderInternalFactory.circularGet (ProviderInternalFactory.java:61)
[2021-09-01T14:44:20.807Z]     at com.google.inject.internal.InternalFactoryToInitializableAdapter.get (InternalFactoryToInitializableAdapter.java:47)
[2021-09-01T14:44:20.807Z]     at com.google.inject.internal.ProviderToInternalFactoryAdapter.get (ProviderToInternalFactoryAdapter.java:40)
[2021-09-01T14:44:20.807Z]     at com.google.inject.internal.SingletonScope$1.get (SingletonScope.java:168)
[2021-09-01T14:44:20.807Z]     at com.google.inject.internal.InternalFactoryToProviderAdapter.get (InternalFactoryToProviderAdapter.java:39)
[2021-09-01T14:44:20.807Z]     at com.google.inject.internal.InjectorImpl$1.get (InjectorImpl.java:1050)
[2021-09-01T14:44:20.807Z]     at org.eclipse.sisu.inject.LazyBeanEntry.getValue (LazyBeanEntry.java:81)
[2021-09-01T14:44:20.807Z]     at org.eclipse.sisu.plexus.LazyPlexusBean.getValue (LazyPlexusBean.java:51)
[2021-09-01T14:44:20.807Z]     at org.eclipse.sisu.wire.EntryListAdapter$ValueIterator.next (EntryListAdapter.java:111)
[2021-09-01T14:44:20.807Z]     at java.util.AbstractCollection.addAll (AbstractCollection.java:343)
[2021-09-01T14:44:20.807Z]     at org.apache.maven.DefaultMaven.getLifecycleParticipants (DefaultMaven.java:397)
[2021-09-01T14:44:20.807Z]     at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:260)
[2021-09-01T14:44:20.807Z]     at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
[2021-09-01T14:44:20.807Z]     at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
[2021-09-01T14:44:20.807Z]     at org.apache.maven.cli.MavenCli.execute (MavenCli.java:957)
[2021-09-01T14:44:20.807Z]     at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:289)
[2021-09-01T14:44:20.807Z]     at org.apache.maven.cli.MavenCli.main (MavenCli.java:193)
[2021-09-01T14:44:20.807Z]     at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
[2021-09-01T14:44:20.807Z]     at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
[2021-09-01T14:44:20.807Z]     at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
[2021-09-01T14:44:20.807Z]     at java.lang.reflect.Method.invoke (Method.java:498)
[2021-09-01T14:44:20.807Z]     at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
[2021-09-01T14:44:20.807Z]     at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
[2021-09-01T14:44:20.807Z]     at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
[2021-09-01T14:44:20.807Z]     at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
[2021-09-01T14:44:20.807Z] Caused by: java.lang.UnsupportedClassVersionError: io/quarkus/maven/components/BootstrapSessionListener has been compiled by a more recent version of the Java Runtime (class file version 55.0), this version of the Java Runtime only recognizes class file versions up to 52.0
[2021-09-01T14:44:20.807Z]
```

I will merge this when receiving Acks.

File: kie-dmn/kie-dmn-core/src/test/java/org/kie/dmn/core/classloader/DMNEvalHelperAccessorCacheTest.java [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+
+        final KieContainer container1 = createKieContainer1(ks);
+        final KieContainer container2 = createKieContainer2(ks);
+
+        checkKieContainer1(container1);
+        checkKieContainer2(container2);
+    }
+
+    private KieContainer createKieContainer1(KieServices ks) {
+        final String javaSource = "package org.acme;\n" +
```
Comment (pibizza) [inline kie-dmn/kie-dmn-core/src/test/java/org/kie/dmn/core/classloader/DMNEvalHelperAccessorCacheTest.java:62]: There is lots of duplication in this and the next method. I would suggest factoring the common code in a parametrized method.

**Review by mariofusco — APPROVED** [review]

File: kie-dmn/kie-dmn-core/src/test/java/org/kie/dmn/core/classloader/DMNEvalHelperAccessorCacheTest.java (hunk 1 above)
Comment (tarilabs) [reply kie-dmn/kie-dmn-core/src/test/java/org/kie/dmn/core/classloader/DMNEvalHelperAccessorCacheTest.java:62]: I believe 9c069f8005 is the a sensible refactor, and I wouldn't push it forward than that on this case.

**Review by hellowdan — APPROVED** [review]

**Review by mariofusco — APPROVED** [review]

**Pull request merged**